server/build_event_protocol/build_status_reporter/build_status_reporter.go

```diff
@@ -261,6 +264,16 @@ func (r *BuildStatusReporter) appURL() string {
 	return r.env.GetConfigurator().GetAppBuildBuddyURL()
 }
 
+func (r *BuildStatusReporter) shouldReportStatusPerTest() bool {
+	shouldReportStatusPerTest := r.env.GetConfigurator().GetGithubConfig().StatusPerTestTarget
+
```

Comment:
Minor nit, but we should ensure that the GithubConfig is non-nil before using it. How about something like?

if githubConfig := r.env.GetConfigurator().GetGithubConfig(); githubConfig != nil {
  if githubConfig.shouldReportStatusPerTest != nil {
    return *githubConfig.shouldReportStatusPerTest
  }
}
return true